samples/third-party/gltf-loader.js

```diff
@@ -0,0 +1,354 @@
+(function() {
+    'use strict';
+
+    var GLTF = {};
+
+    var AttributeSize = {
+        5120: 1, // BYTE
+        5121: 1, // UNSIGNED_BYTE
+        5122: 2, // SHORT
+        5123: 2, // UNSIGNED_SHORT
+        5126: 4  // FLOAT
+    };
+
+    var NumberOfComponents = {
+        'SCALAR': 1,
+        'VEC2': 2,
+        'VEC3': 3,
+        'VEC4': 4,
+        'MAT2': 4,
+        'MAT3': 9,
+        'MAT4': 16
+    };
+
+    // @todo: load multiple scences
+    GLTF.Scene = function() {
+        this.vertices = [];
+        this.indices = [];
+    };
+
+    var parseGltf = function(json, onload) {
+
+        var newScene = new GLTF.Scene;
+
+        // Iterate through every scene
+        for (var sceneName in json.scenes) {
+            var scene = json.scenes[sceneName];
+            var nodes = scene.nodes;
+            var nodeLen = nodes.length;
+
+            // Iterate through every node within scene
+            for (var n = 0; n < nodeLen; ++n) {
+                var nodeName = nodes[n];
+                var node = json.nodes[nodeName];
+
+                // Traverse node
+                traverseNode(json, node, newScene, onload);
+            }
+        }
+    };
+    
+    
+    var translationVec3;
+    var rotationQuat;
+    var scaleVec3;
+    var TRMatrix;
+    
+    var traverseNode = function(json, node, scene, onload, matrix = mat4.create()) {
+
+        var curMatrix = mat4.create();
+        
+        if (node.hasOwnProperty('matrix')) {
+            // matrix
+            for(var i = 0; i < 16; ++i) {
+                curMatrix[i] = node.matrix[i];
+            }
+            mat4.multiply(curMatrix, matrix, curMatrix);
+        } else {
+            // translation, rotation, scale (TRS)
+            // @todo: these labels are optional
```

Comment:
I still need to clean it up, but there is a utility to add default values here: https://github.com/AnalyticalGraphicsInc/gltf-pipeline/blob/master/lib/addDefaults.js